src/Microsoft.Azure.WebJobs.Extensions.Storage/Queues/Listeners/QueueListener.cs

```diff
@@ -377,5 +387,184 @@ private void ThrowIfDisposed()
 
             return queueProcessor;
         }
+
+        public string Id
+        {
+            get
+            {
+                return $"{_functionId}-QueueTrigger-{_queue.Name}".ToLower();
+            }
+        }
+
+        async Task<ScaleMetrics> IScaleMonitor.GetMetricsAsync()
+        {
+            return await GetMetricsAsync();
+        }
+
+        public async Task<QueueTriggerMetrics> GetMetricsAsync()
+        {
+            int queueLength = 0;
```

Comment:
The GetMetricsAsync/GetScaleStatus implementations are a direct port from ScaleController